### PR TITLE
Update LFortran version to latest 0.21.3

### PR DESCRIPTION
**Pull request by Shaikh-Ubaid**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by Shaikh-Ubaid** [comment]
@conda-forge-admin, please rerender

**Comment by Shaikh-Ubaid** [comment]
@conda-forge-admin, please rerender

**Comment by certik** [comment]
Looks like we are missing the `zstd`?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -37,7 +38,6 @@ requirements:
     - nlohmann_json
     - cppzmq
     - zlib
-    - zstd =1.5.2
```
Comment (certik) [inline recipe/meta.yaml:40]: It doesn't work if `zstd-static` is in `host`?

Comment (Shaikh-Ubaid) [reply recipe/meta.yaml:40]: We did not try it yet. I will try it. It looks like currently platforms apart from Linux pass.

**Comment by Shaikh-Ubaid** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -28,6 +28,7 @@ requirements:
     - nlohmann_json     # [build_platform != target_platform]
     - cppzmq            # [build_platform != target_platform]
     - zlib              # [build_platform != target_platform]
+    - zstd-static =1.5.5
```
Comment (certik) [inline recipe/meta.yaml:31]: If the current CI passes, you can then probably remove it from here as well.

**Comment by certik** [comment]
Given how long it takes to run I would just merge this if this passes. We can send a new PR to try to remove the extra dependency per my comment above.

**Pull request merged**